GNUmakefile

```diff
@@ -576,7 +576,11 @@ endif
 deepclean:	clean
 	rm -rf unicorn_mode/unicornafl
 	rm -rf qemu_mode/qemuafl
+ifeq "$(IN_REPO)" "1"
 # NEVER EVER ACTIVATE THAT!!!!! git reset --hard >/dev/null 2>&1 || true
+	git checkout unicorn_mode/unicornafl
+	git checkout qemu_mode/qemuafl
+endif
```

Comment:
@domenukk how is the right action for deepclean being "git checkout"? IMHO this should remove the checkout, but so that it can be recovered again by build_unicorn_...

Reply:
There is no such thing as "downloading a submodule" or "checking out a submodule", they are always part of the repository. They are just not initialized when you do a non-recursive `clone`. Once you initialized and updated the submodule, there is no way to "un-download", although we can reset it to the empty-looking state this way.

TL;DR: Delete changes the node to be deleted in git, not what we want. Checkout checks out the (non-deleted) state of the folder afterwards. It'll still be an empty folder.

The only use I see is for testing.

Reply:
but wouldnt it rather be "make clean ; git stash" or "make clean ; git drop" rather than "git checkout"?

Reply:
What do we want to achieve with deepclean?
If the folders should be empty afterwards(to test the scripts or something?) this is correct. If we want to go back to the original state of the repo, the commented out `reset hard` would have worked.

Make clean and git stash  inside the folders would be a good alternativeas the folder contents are not cleared, but what's the point of deep clean then?

Reply:
originally deepclean was meant to get back to the checkout state, so no checkout of qemuafl and unicorn. (like e.g. you want to create a tar ball from the current repo state). for subversion this was needed for me. but wasnt the "git reset hard" also affecting the whole afl++ repo? that should not be the case I think.

Reply:
If you want to create a tarball, you'd want to do a `git reset hard` to get rid of all temp files and local changes ;)
But if you just want the folders of qemuafl and unicornafl to be emtpy, the current code is correct.